.github/workflows/verify.yml

```diff
@@ -67,10 +67,12 @@ jobs:
 
       - name: Checkout code
         uses: actions/checkout@v2
-        # Required to checkout HEAD^ for the msftidy script
+        # Required to checkout HEAD^ and 3a046f01dae340c124dd3895e670983aef5fe0c5 for the msftidy script
         # https://github.com/actions/checkout/tree/5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f#checkout-head
+        # Note that this adds ~30 seconds to CI time. Msftidy could be extracted to its own job to run
+        # in parallel.
```

Comment:
Extracting msftidy to run in a separate job has the added benefit of running slightly faster overall, but we could in theory automagically write comments letting users know that their linting needs fixed:

```
  - name: Notify user of lint failure
    if: ${{ failure() }}
    uses: actions/github-script@v3
    with:
      github-token: ${{secrets.GITHUB_TOKEN}}
      script: |
            github.issues.addLabels({
              issue_number: context.issue.number,
              owner: context.repo.owner,
              repo: context.repo.repo,
              labels: ['needs-linting']
            })

  - name: Notify user of lint success
    if: ${{ failure() }}
    uses: actions/github-script@v3
    with:
      github-token: ${{secrets.GITHUB_TOKEN}}
      script: |
            github.issues.addLabels({
              issue_number: context.issue.number,
              owner: context.repo.owner,
              repo: context.repo.repo,
              name: 'needs-linting'
            })
```

If bots trigger bots that is, will investigate 👀

Reply:
Looks like actions can trigger bots on pull requests where the branch is from the same repo

![image](https://user-images.githubusercontent.com/60357436/107453146-54397f80-6b42-11eb-936c-163cef3eaea7.png)

Will have to investigate the permissions of third party pull requests.

Reply:
Looks like using the new `pull_request_target` event wouldn't quite work as we want to use the workflow from master, but the code from a branch - which would expose the github token with write access as an environment variable. Will investigate further

https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/

Reply:
This is definitely possible, but will extract that effort from this PR for now:

<details>
<summary>WIP changes</summary>

```yaml
name: Lint

on:
  push:
    branches-ignore:
      - gh-pages
      - metakitty
  # Note this is 'pull_request_target' and _not_ 'pull_request' for security reasons of having write tokens:
  #   https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/
  # This means that a pull request's workflow will not run, but rather the one that's been committed to master.
  pull_request_target:
    branches:
      - '*'
    types: [synchronize, reopened]

jobs:
  test:
    runs-on: ubuntu-16.04
    timeout-minutes: 40

    strategy:
      fail-fast: true
      matrix:
        ruby:
          - 2.5

    name: Lint ${{ matrix.ruby }}
    steps:
     # ...

      - name: Notify user of lint success
        if: ${{ success() }}
        uses: actions/github-script@v3
        with:
          github-token: ${{secrets.GITHUB_TOKEN}}
          script: |
            github.issues.removeLabel({
              issue_number: context.issue.number,
              owner: context.repo.owner,
              repo: context.repo.repo,
              labels: ['needs-linting']
            })

      - name: Notify user of lint failure
        if: ${{ failure() }}
        uses: actions/github-script@v3
        with:
          github-token: ${{secrets.GITHUB_TOKEN}}
          script: |
            github.issues.addLabels({
              issue_number: context.issue.number,
              owner: context.repo.owner,
              repo: context.repo.repo,
              labels: ['needs-linting']
            })

```
</details>